File: backend/app/schemas/document.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.document import DocumentStatus


class DocumentPublic(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    course_id: int
    week_id: int | None
    filename: str
    file_type: str
    status: DocumentStatus
    page_count: int | None
    pages_needing_ocr: int
    chunk_count: int
    error_message: str | None
    created_at: datetime


class ChunkSearchResult(BaseModel):
    document_id: int
    document_filename: str
    page_number: int | None
    content: str
    similarity: float


class ChunkSearchRequest(BaseModel):
    query: str
    course_id: int | None = None
    top_k: int = 5

File: frontend/src/types/Document.ts
export type DocumentStatus = 'pending' | 'processing' | 'ready' | 'failed'

export interface DocumentPublic {
  id: number
  course_id: number
  week_id: number | null
  filename: string
  file_type: string
  status: DocumentStatus
  page_count: number | null
  pages_needing_ocr: number
  chunk_count: number
  error_message: string | null
  created_at: string
}

export interface ChunkSearchResult {
  document_id: number
  document_filename: string
  page_number: number | null
  content: string
  similarity: number
}
